csvjoin/functions.py

```diff
@@ -0,0 +1,257 @@
+        splits by comma.
+
+        :param line: The line to split and rstrip.
+        :type line: str
+        :return: The array of strings.
+        :rtype: [str]
+   """
+    new_line = str(line)
+    new_line = new_line.rstrip()
+    return new_line.split(',')
+
+
+def print_non_inner_join(file, header: str, taken_rows: set, join_type: str, column_index: int):
+    """
+        Prints non-inner type of join in order to "fill" NULL values.
+
+        :param file: The handle of the file.
+        :param header: The header of the second file to get the
+        number of the NULLs.
+        :param taken_rows: The set of already printed rows of the file.
+        :param join_type: The type of the join.
+        :param column_index: The column_index to pass it
+        to the parse_data_line function.
+        :type file: TextIO
+        :type header: str
+        :type taken_rows: set
+        :type join_type: str
+        :type column_index: int
+    """
+    file.seek(0, 0)
+    file.readline()
+    line = file.readline()
+
+    # creating NULL line string
+    header = header.split(',')
+    n = len(header)
+    filler = ''
+    for _ in range(n - 1):
+        filler += 'NULL'
+        filler += ','
+    filler += 'NULL'
+
+    counter = 1
+    while line != '':
+        if counter in taken_rows:
+            line = file.readline()
+            counter += 1
+            continue
+
+        if join_type == 'left':
+            parse_data_line(line, filler, -1, column_index)
+
+        if join_type == 'right':
+            parse_data_line(filler, line, column_index, -1)
+
+        line = file.readline()
+        counter += 1
+
+
+def join_files(file_path1: str, file_path2: str, column_name: str, join_type: str):
+    """
+        The actual main algorithm for printing the joined files.
+
+        :param file_path1: The path of the first file.
+        :param file_path2: The path of the second file.
+        :param column_name: The name of the column to join on.
+        :param join_type: The type of the join.
+        :type file_path1: str
+        :type file_path2: str
+        :type column_name: str
+        :type join_type: str
+    """
+    try:
+        # Even though I can have left or right join
+        # firstly I need to do inner and after that
+        # I can do the rest if needed
+        with open(file_path1) as file1, open(file_path2) as file2:
+            header1 = str(file1.readline())
+            header2 = str(file2.readline())
+
+            # look for the index where column_name appears in header of the CSV files
+            column_index1 = find_index_of_column(header1, column_name)
+            column_index2 = find_index_of_column(header2, column_name)
+
+            if join_type == 'right':
+                parse_data_line(header1, header2, column_index1, -1)
+            else:
+                parse_data_line(header1, header2, -1, column_index2)
+
+            taken_rows1 = set()
+            taken_rows2 = set()
```

Comment:
I am pretty sure that results of `JOIN` can be too large for memory, based on task description. This would not work, you would need CSV writer to write results in a separate temporary file, using `tempfile` module, for example.